[supervisord]
nodaemon=true
user=root
logfile=/var/log/supervisor/supervisord.log
pidfile=/var/run/supervisord.pid
childlogdir=/var/log/supervisor/

[program:postgresql]
command=/usr/lib/postgresql/15/bin/postgres -D /var/lib/postgresql/data -c config_file=/var/lib/postgresql/data/postgresql.conf
user=postgres
autostart=true
autorestart=true
priority=100
stdout_logfile=/var/log/supervisor/postgresql.log
stderr_logfile=/var/log/supervisor/postgresql_error.log
environment=PGDATA=/var/lib/postgresql/data

[program:redis]
command=redis-server --daemonize no --protected-mode no --bind 0.0.0.0
user=mev
autostart=true
autorestart=true
priority=90
stdout_logfile=/var/log/supervisor/redis.log
stderr_logfile=/var/log/supervisor/redis_error.log

[program:django]
command=python manage.py runserver 0.0.0.0:8000
directory=/app
user=mev
autostart=true
autorestart=true
priority=80
stdout_logfile=/var/log/supervisor/django.log
stderr_logfile=/var/log/supervisor/django_error.log
environment=DJANGO_SETTINGS_MODULE=mev_dashboard.settings,PYTHONPATH=/app,PATH=/opt/venv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

[program:nginx]
command=nginx -g "daemon off;"
user=root
autostart=true
autorestart=true
priority=70
stdout_logfile=/var/log/supervisor/nginx.log
stderr_logfile=/var/log/supervisor/nginx_error.log

[program:django-init]
command=/app/init_django.sh
directory=/app
user=mev
autostart=true
autorestart=false
priority=110
startsecs=0
startretries=1
stdout_logfile=/var/log/supervisor/django_init.log
stderr_logfile=/var/log/supervisor/django_init_error.log
environment=DJANGO_SETTINGS_MODULE=mev_dashboard.settings,PYTHONPATH=/app,PATH=/opt/venv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

[group:web]
programs=django,nginx
priority=999

[group:database]
programs=postgresql,redis
priority=1000

[group:all]
programs=django-init,postgresql,redis,django,nginx
priority=1000